create.ps1

```diff
@@ -1,35 +1,190 @@
+    }
+    #endregion Verify correlation configuration and properties
+
+    #region Import CSV data
+    $actionMessage = "importing data from CSV file at path [$($actionContext.Configuration.CsvPath)]"
+    
+    $csvContent = $null
+    $csvContent = Import-Csv -Path $actionContext.Configuration.CsvPath -Delimiter $actionContext.Configuration.Delimiter -Encoding $actionContext.Configuration.Encoding
+    
+    # Group on correlation field to match employee to CSV row(s)
+    $csvContentGrouped = $csvContent | Group-Object -Property $correlationField -AsString -AsHashTable
+
+    Write-Verbose "Imported data from CSV file at path [$($actionContext.Configuration.CsvPath)]. Result count: $(($csvContent | Measure-Object).Count)"
+    #endregion Import CSV data
+
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        #region Get current row for person
+        $actionMessage = "querying CSV row where [$($correlationField)] = [$($correlationValue)]"
+        
+        $currentRow = $null
+        if ($csvContentGrouped -ne $null) {
+            $currentRow = $csvContentGrouped["$($correlationValue)"]
+        }
+
+        Write-Verbose "Queried CSV row where [$($correlationField)] = [$($correlationValue)]. Result count: $(($currentRow | Measure-Object).Count)"
+        #endregion Get current row for person
+    }
+
+    #region Account
+    #region Calulate action
+    $actionMessage = "calculating action"
+    if (($currentRow | Measure-Object).count -eq 0) {
+        $action = "Create"
+    }
+    elseif (($currentRow | Measure-Object).count -eq 1) {
+        $action = "Correlate"
+    }
+    elseif (($currentRow | Measure-Object).count -gt 1) {
+        $action = "MultipleFound"
+    }
+    #endregion Calulate action
+
+    #region Process
+    switch ($action) {
+        "Create" {
+            #region Create csv row
+            $actionMessage = "creating row in CSV"
+
+            #region Create custom updated csv object
+            $updatedCsvContent = $null
+            $updatedCsvContent = [System.Collections.ArrayList](, ($csvContent))
+            #endregion Create custom updated csv object
+
+            #region Add new CSV row to custom updated csv object
+            [void]$updatedCsvContent.Add($account)
+            #endregion Add new CSV row to custom updated csv object
+
+            #region Export updated CSV object
+            $exportCsvSplatParams = @{
+                Path              = $actionContext.Configuration.CsvPath
+                Delimiter         = $actionContext.Configuration.Delimiter
+                Encoding          = $actionContext.Configuration.Encoding
+                NoTypeInformation = $true
+                ErrorAction       = "Stop"
+                Verbose           = $false
+            }
+
+            if (-Not($actionContext.DryRun -eq $true)) {
+                Write-Verbose "SplatParams: $($exportCsvSplatParams | ConvertTo-Json)"
```

Comment:
Do we really need to log the parameters for `Export-Csv`? Since they are hardcoded or defined in the configuration.